### PR TITLE
logictest: fix bogus UTC normalization causing race error

### LINKED ISSUE #40336: sql: running logic tests with stressrace produces race in time.Now()
When I run logic tests with `stressrace`, I sometimes get a failure due to a data race in `time.Now()`. This seems to be the exact same problem as #30027, which was thought to have been fixed as of Go 1.11 (I think), but I'm getting it with Go 1.12.

The most reliable way I've found to reproduce this is by supplying a `TESTS` regex that doesn't match any tests (e.g., `TestLogic/not_a_real_test`), in which case I get this failure every time:
```
lucy@gceworker-lucy:~/go/src/github.com/cockroachdb/cockroach$ go version
go version go1.12.7 linux/amd64
lucy@gceworker-lucy:~/go/src/github.com/cockroachdb/cockroach$ make stressrace TESTS='TestLogic/not_a_real_test' PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m'
Running make with -j24
GOPATH set to /home/lucy/go
Cleaning old generated files.
GOFLAGS= go test -race -exec 'stress -maxtime 20m -timeout 10m' -tags ' gss make x86_64_linux_gnu' -ldflags '-X github.com/cockroachdb/cockroach/pkg/build.typ=development -extldflags "" -X "github.com/cockroachdb/cockroach/pkg/build.tag=v19.2.0-alpha.20190606-1687-gcfde7a92b5-dirty" -X "github.com/cockroachdb/cockroach/pkg/build.rev=cfde7a92b551ae02599df33a46c44c8ec266a3c6" -X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=x86_64-linux-gnu"  ' -run "TestLogic/not_a_real_test" -timeout 0 github.com/cockroachdb/cockroach/pkg/sql/logictest  -v -args -test.timeout 5m

I190829 17:43:12.168269 1 rand.go:83  Random seed: -5711090402640773854
=== RUN   TestLogic
--- total: 0 tests, 0 failures
--- PASS: TestLogic (0.18s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestLogic318149004
    test_log_scope.go:58: use -show-logs to present logs inline
testing: warning: no tests to run
PASS
==================
WARNING: DATA RACE
Read at 0x000007b297e8 by goroutine 9:
  time.Now()
      /usr/local/go/src/time/time.go:1094 +0xf5
  time.sendTime()
      /usr/local/go/src/time/sleep.go:138 +0x44

Previous write at 0x000007b297e8 by goroutine 42:
  [failed to restore the stack]

Goroutine 9 (running) created at:
  runtime.(*timersBucket).addtimerLocked()
      /usr/local/go/src/runtime/time.go:169 +0x10d
  github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
      /usr/local/go/src/time/tick.go:58 +0x42

Goroutine 42 (finished) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1072 +0x2eb
  github.com/cockroachdb/cockroach/pkg/sql/logictest.TestMain()
      /home/lucy/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/main_test.go:33 +0x15d
  main.main()
      _testmain.go:46 +0x222
==================


ERROR: exit status 66

1 runs completed, 1 failures, over 1s
context canceled
FAIL
FAIL	github.com/cockroachdb/cockroach/pkg/sql/logictest	1.155s
Makefile:962: recipe for target 'stressrace' failed
make: *** [stressrace] Error 1
```

This also happens sometimes if I run other actual tests, though - simply running all the logic tests usually causes this to happen.

### PR DESCRIPTION
**Pull request by cucaroach**
Fixes #40336

### CODE REVIEW

**Comment by cucaroach** [comment]
I need to beef up the commit notes and figure out how to do the same for bazel but curious what you think about this approach @otan

File: Makefile [hunk 1]
```diff
@@ -220,8 +220,10 @@ LDFLAGS ?=
 CGO_CFLAGS = $(filter-out -g%,$(CFLAGS))
 CGO_CXXFLAGS = $(CXXFLAGS)
 CGO_LDFLAGS = $(filter-out -static,$(LDFLAGS))
+# certain time based fail if UTC isn't the default timezone
```
Comment (otan) [inline Makefile:223]: nit: `Certain time based operations fail`

Can you describe what this approach is? I'm always hesitant to add a new global. Might be worth consulting dev-inf.

Comment (cucaroach) [reply Makefile:223]: We have lots of timestamp tests that rely on the local timezone being UTC.   This was achieved by writing to the time.Local to force it to be UTC.   I changed it to achieve that via an env var instead.   Here are the relevant Go docs: https://golang.org/src/time/zoneinfo.go?#L78

Comment (otan) [reply Makefile:223]: ah its just an env var. this seems reasonable to me.

Comment (cucaroach) [reply Makefile]: Cool, who can I ask about how to do this for bazel?

Comment (otan) [reply Makefile]: dev-inf

**Comment by cucaroach** [comment]
@otan can you help me to diagnose the bazel failures I'm seeing here, I can't make heads or tails of it, looks like a timeout issue?

**Comment by cucaroach** [comment]
Nevermind, looks like a flake!

**Review by cucaroach — COMMENTED** [review]
RFAL SVP

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64482#-:-Mc7L2O_E5bZd0a9wVua:b-lg9nn3)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[Makefile, line 223 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/64482#-MaEM6DKpLxU3tGk7Pjl-r1-223:-Mc7LRhq8hGpSCUwgAwS:b-8fx2u7) ([raw file](https://github.com/cockroachdb/cockroach/blob/9bca94368a6968e83d3443ba536eb80cd87a8288/Makefile#L223)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

dev-inf
</blockquote></details>

:+1:

**Review by rail — APPROVED** [review]
LGTM, :shipit:

**Comment by cucaroach** [comment]
TFTR!  bors r+

**Pull request merged**